core/contract.py

```
@@ -0,0 +1,133 @@
+    """
+    Worker function to execute contract code in a separate process.
+    """
+    try:
+        # Attempt to set resource limits (Unix only)
+        try:
+            import resource
+            # Limit CPU time (seconds) and memory (bytes) - example values
+            resource.setrlimit(resource.RLIMIT_CPU, (1, 1))
+            # resource.setrlimit(resource.RLIMIT_AS, (100 * 1024 * 1024, 100 * 1024 * 1024))
+        except ImportError as e:
+            raise RuntimeError(f"Resource limits not enforced: {e}")
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)